code/modules/vore/eating/belly_vr.dm

```diff
@@ -389,6 +393,57 @@
 	var/strsound = struggle_sounds[strpick]
 	playsound(R.loc, strsound, 50, 1)
 
+	if(escapable) //If the stomach has escapable enabled.
+		R << "<span class='warning'>You attempt to climb out of \the [name].</span>"
+		owner << "<span class='warning'>Someone is attempting to climb out of your [name]!</span>"
+		if(prob(escapechance)) //Let's have it check to see if the prey escapes first.
+			if(do_after(R, escapetime))
+				if((escapable) && (R in internal_contents)) //Does the owner still have escapable enabled?
+					release_specific_contents(R)
+					R << "<span class='warning'>You climb out of \the [name].</span>"
+					owner << "<span class='warning'>[R] climbs out of your [name]!</span>"
+					for(var/mob/M in hearers(4, owner))
+						M.show_message("<span class='warning'>[R] climbs out of [owner]'s [name]!</span>", 2)
+					return
+				else if(!(R in internal_contents)) //Aren't even in the belly. Quietly fail.
+					return
+			else //Belly became inescapable.
+				R << "<span class='warning'>Your attempt to escape [name] has failed!</span>"
+				owner << "<span class='notice'>The attempt to escape from your [name] has failed!</span>"
+				return
+
+		else if(prob(transferchance)) //Next, let's have it see if they end up getting into an even bigger mess then when they started.
+			var/datum/belly/T = transferlocation
+			for(var/K in owner.vore_organs)
+				var/datum/belly/B = owner.vore_organs[K]
+				var/datum/belly/TL = B.transferlocation
+				if(TL != null)
+					B.internal_contents -= R
+					T.internal_contents += R
+					R << "<span class='warning'>Your attempt to escape [name] has failed and your struggles only results in you sliding into [owner]'s [transferlocation]</span>"
+					owner << "<span class='warning'>Someone slid into your [transferlocation] due to their struggling inside your [name]!</span>"
+					return
+
+		else if(prob(absorbchance)) //After that, let's have it run the absorb chance.
+			R << "<span class='warning'>In responce to your struggling, \the [name] begins to get more active...</span>"
+			owner << "<span class='warning'>You feel your [name] beginning to become active!</span>"
+			for(var/K in owner.vore_organs)
+				var/datum/belly/B = owner.vore_organs[K]
+				B.digest_mode = DM_ABSORB
+				return
+
+		else if(prob(digestchance)) //Finally, let's see if it should run the digest chance.
+			R << "<span class='warning'>In responce to your struggling, \the [name] begins to get more active...</span>"
```

Comment:
*response

Also is there a check to see if we're already digesting? Else this gets spammy.